test/test_autograd.py

```diff
@@ -2299,6 +2299,29 @@ def func(root, b, upper):
                                                [True, False]):
             _test_with_size(a_size, b_size, upper)
 
+    @skipIfNoLapack
+    def test_eig(self):
+        def func(B):
+            return torch.eig(B, eigenvectors=True)
+
+        def run_test(dims):
+            # The backward operation for eig only works for real eigenvalues,
+            # so the matrix should be B = U^{-1}*A*U where A is a random
+            # symmetric matrix and U is a random full-rank matrix.
+            # Slight change to the matrix should not make the eigenvalues
+            # complex, so we apply requires_grad_ to B, not A and U
+
+            A = random_symmetric_matrix(dims[-1], *dims[:-2])
+            U = torch.rand(*dims)
+            Uinv = torch.inverse(U)
+            B = torch.matmul(Uinv, torch.matmul(A, U)).requires_grad_()
+
+            gradcheck(func, [B])
+            gradgradcheck(func, [B])
+
+        for dims in [(3, 3), (5, 5)]:
+            run_test(dims)
```

Comment:
Could you add a test here where only one of the two outputs sends gradients?
In particular, to make sure the logic if one grad are not defined is working fine.

Reply:
Sure, added them in the new commit.